tests/test_modeling_tf_wav2vec2.py

```diff
@@ -0,0 +1,349 @@
+        self.feat_extract_norm = feat_extract_norm
+        self.feat_extract_dropout = feat_extract_dropout
+        self.feat_extract_activation = feat_extract_activation
+        self.conv_dim = conv_dim
+        self.conv_stride = conv_stride
+        self.conv_kernel = conv_kernel
+        self.conv_bias = conv_bias
+        self.num_conv_pos_embeddings = num_conv_pos_embeddings
+        self.num_conv_pos_embedding_groups = num_conv_pos_embedding_groups
+        self.num_hidden_layers = num_hidden_layers
+        self.num_attention_heads = num_attention_heads
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.intermediate_size = intermediate_size
+        self.layer_norm_eps = layer_norm_eps
+        self.hidden_act = hidden_act
+        self.initializer_range = initializer_range
+        self.vocab_size = vocab_size
+        self.do_stable_layer_norm = do_stable_layer_norm
+        self.scope = scope
+
+        output_seq_length = self.seq_length
+        for kernel, stride in zip(self.conv_kernel, self.conv_stride):
+            output_seq_length = (output_seq_length - (kernel - 1)) / stride
+        self.output_seq_length = int(math.ceil(output_seq_length))
+        self.encoder_seq_length = self.output_seq_length
+
+        self.pt_model = Wav2Vec2ForCTC.from_pretrained("facebook/wav2vec2-base-960h")
+        self.tf_model = TFWav2Vec2ForCTC.from_pretrained(MODEL_PATH)
+
+    def prepare_config_and_inputs(self):
+        input_ids = tf.cast(ids_tensor([self.batch_size, self.seq_length], 32768), tf.float32) / 32768.0
+
+        config = Wav2Vec2Config(
+            hidden_size=self.hidden_size,
+            feat_extract_norm=self.feat_extract_norm,
+            feat_extract_dropout=self.feat_extract_dropout,
+            feat_extract_activation=self.feat_extract_activation,
+            conv_dim=self.conv_dim,
+            conv_stride=self.conv_stride,
+            conv_kernel=self.conv_kernel,
+            conv_bias=self.conv_bias,
+            num_conv_pos_embeddings=self.num_conv_pos_embeddings,
+            num_conv_pos_embedding_groups=self.num_conv_pos_embedding_groups,
+            num_hidden_layers=self.num_hidden_layers,
+            num_attention_heads=self.num_attention_heads,
+            hidden_dropout_prob=self.hidden_dropout_prob,
+            intermediate_size=self.intermediate_size,
+            layer_norm_eps=self.layer_norm_eps,
+            hidden_act=self.hidden_act,
+            initializer_range=self.initializer_range,
+            vocab_size=self.vocab_size,
+        )
+
+        inputs_dict = {"input_ids": input_ids}
+        return config, inputs_dict
+
+    def create_and_check_model(self, config, input_dict):
+        model = TFWav2Vec2Model(config)
+        result = model(input_dict)
+        self.parent.assertEqual(
+            result.last_hidden_state.shape, (self.batch_size, self.output_seq_length, self.hidden_size)
+        )
+
+    def check_pos_embedding(self, config, input_dict):
```

Comment:
we usually just have a single PT<=>TF conversion test which already seems to work under `test_pt_tf_model_equivalence`. All of these tests should also be run when PyTorch is not installed.

Reply:
@patrickvonplaten I'm taking `check_pos_embedding` out, but I found a small bug that is related to this specific test. I have to manually set the weights of the TensorFlow layer from the Pytorch layer to get this test to pass. So in `wrice/wav2vec2-base-960h`, I manually did this before I exported the model. The weight that doesn't match is `weight_v`, however, `weight_g` and `bias` match 100%. The problem is that I'm not exactly sure why `weight_v` is not getting set correctly. Would [this](https://github.com/huggingface/transformers/blob/77f4c46b501322e9bffb5416dfbf0397deefd7d8/src/transformers/modeling_tf_utils.py#L454) be the proper place to start debugging this?

Reply:
ah yeah there might be an issue with the conversion there actually since the weight naming is very weird -> I look into it :-)